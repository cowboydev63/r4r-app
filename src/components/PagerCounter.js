import React from 'react';
import PropTypes from 'prop-types';

class PagerCounter extends React.PureComponent {
    static propTypes = {
        from: PropTypes.number.isRequired,
        to: PropTypes.number.isRequired,
        total: PropTypes.number.isRequired,
    }

    render() {
        if(!this.props.from || !this.props.to || !this.props.total) {
            return null;
        }
        return (
            <div className="r4r-pager__count" aria-label="Results count">
                <p>{`${ this.props.from } - ${ this.props.to } of ${ this.props.total }`}</p>
            </div>
        )
    }
}

export default PagerCounter;