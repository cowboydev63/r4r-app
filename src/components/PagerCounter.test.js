import React from 'react';
import { shallow } from 'enzyme';
import PagerCounter from './PagerCounter';

describe('PagerCounter Component', () => {
  it('renders without crashing', () => {
    shallow(<PagerCounter from={0} to={0} total={0} />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<PagerCounter from={0} to={20} total={100} />);
    expect(wrapper).toMatchSnapshot();
  })
})