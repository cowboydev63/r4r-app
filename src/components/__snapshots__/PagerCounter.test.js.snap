// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagerCounter Component renders correctly 1`] = `""`;
